### PR TITLE
✨Source Airtable: new stream `Collaborators`

### LINKED ISSUE #23896: Source Airtable: new stream Collaborators
## Tell us about the problem you're trying to solve
[Collaborators](https://airtable.com/developers/web/api/get-base-collaborators)

## Describe the solution you’d like
Research pros and cons for implementing this stream

### PR DESCRIPTION
**Pull request by gittysachin**
## What
Closes #23896 

## Enhancement
New streams:

* Collaborators

## Recommended reading order
1. `streams.py`
2. `source.py`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Please let me know if my comment makes sense! Thanks for the contribution.

File: airbyte-integrations/connectors/source-airtable/source_airtable/streams.py [hunk 1]
```diff
@@ -155,3 +174,12 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
...
         return self.stream_path
+
+
+class AirtableStreamFactory:
+    @staticmethod
+    def get_stream(stream_path: str, stream_name: str, stream_schema, authenticator) -> AirtableStream:
+        if stream_name == "collaborators":
+            return AirtableCollaborators(stream_path=stream_path, stream_name=stream_name, stream_schema=stream_schema, authenticator=authenticator)
+        else:
+            return AirtableStream(stream_path=stream_path, stream_name=stream_name, stream_schema=stream_schema, authenticator=authenticator)
```
[the remark this answers is not in the document]
Comment (gittysachin) [reply airbyte-integrations/connectors/source-airtable/source_airtable/streams.py:185]: It's not used right now but I thought can be used later.

File: airbyte-integrations/connectors/source-airtable/source_airtable/source.py [hunk 2]
```diff
@@ -97,9 +107,17 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
...
+                    stream_schema=stream["stream"].json_schema,
+                    authenticator=self._auth,
+                )
+            else:
+                yield AirtableStream(
+                    stream_path=stream["stream_path"],
+                    stream_name=stream["stream"].name,
+                    stream_schema=stream["stream"].json_schema,
+                    authenticator=self._auth,
+                )
```
[the remark this answers is not in the document]
Comment (gittysachin) [reply airbyte-integrations/connectors/source-airtable/source_airtable/source.py:123]: Got it, will change

File: airbyte-integrations/connectors/source-airtable/source_airtable/streams.py [hunk 3]
```diff
@@ -88,6 +88,25 @@ def path(self, **kwargs) -> str:
...
+    def path(self, **kwargs) -> str:
+        return f"meta/bases/{self.base_id}"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping[str, Any]]:
+        data = response.json()
+        collaborators = data.get("collaborators", {})
+        base_collaborators = collaborators.get("baseCollaborators", [])
+        workspace_collaborators = collaborators.get("workspaceCollaborators", [])
+        yield from base_collaborators
+        yield from workspace_collaborators
```
[the remark this answers is not in the document]
Comment (gittysachin) [reply airbyte-integrations/connectors/source-airtable/source_airtable/streams.py:107]: Sure

File: airbyte-integrations/connectors/source-airtable/source_airtable/streams.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-airtable/source_airtable/streams.py:185]: If its not used remove it. When you need add the implementation.

**Comment by gittysachin** [comment]
Will raise PRs with correct commits

**Pull request closed without merging**